### ISSUE TITLE
[Bug]: Why RTL is not working in tiptap editor?

### ISSUE THREAD

**Issue by coderyeasin** [ueberdosis/tiptap#3957]
### Which packages did you experience the bug in?

core

### What Tiptap version are you using?

2.0.0

### What’s the bug you are facing?

 when I applied RTL and it should take RTL styles but its not. Otherwise, tiptap can't recognize which one "LTR"/"RTL"

### What browser are you using?

Chrome

### What did you expect to happen?

Hi there,
I think one of the most popular editors tiptap in react js. why missing RTL? Please add RTL support in tiptap editor, it would be great to get more users. `dir="auto"` is not a solution is a take us bluff. when I style something on RTL it's not applied because of the `'auto'` direction, it cannot catch 'RTL' or 'LTR'. so `dir='auto'` is not a good solution, either you make an extension for that.
OR
If you guys are already implemented/manage somehow to solve this problem (like: above problem) please let me know, and I will implement that.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by Bunkerbewohner** [comment]
Out of the box it seems TipTap indeed doesn't deal with RTL text. Have you seen this closed issue already? https://github.com/ueberdosis/tiptap/issues/116

@amirhhashemi wrote an extension to address this: https://github.com/amirhhashemi/tiptap-text-direction
I haven't tested it myself, but I hope it works. 

The [Lexical](https://lexical.dev) editor library seems to use a similar approach. There it just works out of the box. They literally add `dir="ltr"` or `dir="rtl"` to every node, depending on its content.

It would probably make sense for something like this to be included in TipTap's core, considering that there are around 650M people natively speaking (and writing) RTL languages like Arabic, Hebrew and Urdu. Maybe some TitTap contributor will find the time some day. :)

**Comment by alirezaalavi87** [comment]
> The [Lexical](https://lexical.dev/) editor library seems to use a similar approach. There it just works out of the box. They literally add dir="ltr" or dir="rtl" to every node, depending on its content.

I don't think this is the solution. with `dir=rtl`, BiDi (Bi Directional) text would be rendered wrong. I don't know what @coderyeasin means by saying `dir="auto"` doesn't work but in my experience, this is the way to make BiDi text work.

RTL only text already renders ok in tiptap. It doesn't start from the right end of the textbox, but it is still usable. The problem is with BiDi text:

one دو three

یک two سه

As you can see, github handles these ok because it adds `dir=auto` to every `<p>` element. This is the only needed think IMO and it's how most BiDi extensions I've seen work.

**Comment by bdbch** [comment]
Would this dir attribute be required on **all** elements inside the editor or just paragraphs?

**Comment by amirhhashemi** [comment]
> Would this dir attribute be required on **all** elements inside the editor or just paragraphs?

The direction of an element is inherited from its parent. For example, if an element has `dir="rtl"`, its children are also RTL, unless an element explicitly overrides it. So technically, you don't need to add the `dir` attribute to **all** elements. When I build a RTL website, I just add `dir="rtl"` to the `<html>` element and don't add the `dir` attribute to any other element unless I want it to be LTR.

However, for TipTap, I think it makes more sense to add the `dir` attribute to all elements by default. This makes the direction of the content consistent regardless of whether it's shown in a RTL or LTR website. This is specially important for multilingual websites. You can then add an option that disables this behaviour ti add the `dir` attribute only when necessary, for those who are not building a multilingual website.

I also want to highlight that you probably want add the `dir` attribute to other elements besides paragraphs. For example, the placement of the marker of the `<li>` element is based on the direction of the `<li>` element itself. If you add the `dir` attribute only to the `<p>` element inside of the `<li>` element, the marker would be shown on the wrong side.

I highly recommend checking out the [Lexical source code](https://github.com/facebook/lexical). They obviously care a lot about this. [My plugin](https://github.com/amirhhashemi/tiptap-text-direction) is also based on Lexical.
